### ISSUE TITLE
Azure vm module needs higher role for first start/stop action

### ISSUE THREAD

**Issue by guillemsola** [ansible/ansible#41003]
##### SUMMARY

The basic role to start/stop a machine in Azure is _DevTest Labs User_. With the Ansible module the first time a machine is started a _Contributor_ role is needed. Notice that after this "first time" the _DevTest Labs User_ can be used.

##### ISSUE TYPE
 - Bug Report

##### ANSIBLE VERSION
```
ansible 2.5.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/vmadmin/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
```

```
azure-cli-core               2.0.33
azure-cli-nspkg              3.0.2
azure-common                 1.1.11
azure-mgmt-batch             4.1.0
azure-mgmt-compute           2.1.0
azure-mgmt-containerinstance 0.4.0
azure-mgmt-containerregistry 2.0.0
azure-mgmt-containerservice  2.0.0
azure-mgmt-dns               1.2.0
azure-mgmt-keyvault          0.40.0
azure-mgmt-network           1.7.1
azure-mgmt-nspkg             2.0.0
azure-mgmt-rdbms             0.2.0rc1
azure-mgmt-resource          1.2.2
azure-mgmt-sql               0.7.1
azure-mgmt-storage           1.5.0
azure-mgmt-web               0.32.0
azure-nspkg                  2.0.0
azure-storage                0.35.1
```

##### OS / ENVIRONMENT
```
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=16.04
DISTRIB_CODENAME=xenial
DISTRIB_DESCRIPTION="Ubuntu 16.04.4 LTS"
```

##### STEPS TO REPRODUCE

- Deploy  a machine using the Azure portal
- Create a service principal account `az ad sp create-for-rbac --role "DevTest Labs User"`
- Machine is stopped
- Try to open machine with 
```
---
- hosts: localhost
  connection: local
  gather_facts: false

  tasks:
  - name: Scenario 1
    azure_rm_virtualmachine:
      resource_group: "{{ resource_group }}"
      name: "{{ item.name }}"
      allocated: "{{ item.status }}"
    loop:
      - { name: WSLBack-1, status: yes }
```

##### EXPECTED RESULTS

The _DevTest Labs User_ role selected is able to start the machine the first time machine it is started.

##### ACTUAL RESULTS

The first time the machine is started the principal user using it needs a user '_Contributor_' role. After it has been started the _DevTest Labs User_ is enough to start the machine.

**Comment by guillemsola** [comment]
We found this issue while doing tests for #39958. It seems that there are some weird behaviors that happens only once using Ansible and Azure. Maybe they have some kind relation under the hoods.

**Comment by jeffmatts** [comment]
Hello. I'm having a similar issue as in [#39958](https://github.com/ansible/ansible/issues/39958) with Ansible 2.5.5. The below error occurs regardless of which role my service principal has. I've tried both Contributor and Owner. 

>  fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "No handlers could be found for logger
 \"msrestazure.azure_active_directory\"\nTraceback (most recent call last):\n  File \"/tmp/ansible_zX2JPz
/ansible_module_azure_rm_virtualmachine.py\", line 1694, in <module>\n    main()\n  File \"/tmp/ansible_z
X2JPz/ansible_module_azure_rm_virtualmachine.py\", line 1690, in main\n    AzureRMVirtualMachine()\n  Fil
e \"/tmp/ansible_zX2JPz/ansible_module_azure_rm_virtualmachine.py\", line 714, in __init__\n    supports_
check_mode=True)\n  File \"/tmp/ansible_zX2JPz/ansible_modlib.zip/ansible/module_utils/azure_rm_common.py
\", line 345, in __init__\n  File \"/tmp/ansible_zX2JPz/ansible_module_azure_rm_virtualmachine.py\", line
 1076, in exec_module\n    storage_account_type=vm_dict['properties']['storageProfile']['osDisk']['manage
dDisk']['storageAccountType']\nKeyError: 'storageAccountType'\n", "module_stdout": "", "msg": "MODULE FAI
LURE", "rc": 1} 

```
- name: VM State
      azure_rm_virtualmachine:
        resource_group: test
        name: VM1
        allocated: yes
```

**Comment by guillemsola** [comment]
@jtoddmatt I suggest the workaround

1. start VM from portal/powershell
2. power off with Ansible

After this you should be able to start them back with Ansible.

**Comment by Fred-sun** [comment]
@zikalino Could you take a look this issue? Thanks!

**Comment by Fred-sun** [comment]
@zikalino Could you take a look this issue? Thanks!

**Comment by Fred-sun** [comment]
@zikalino Could you take a look this issue? Thanks！

**Comment by Fred-sun** [comment]
@zikalino This issue similar with [#39958](https://github.com/ansible/ansible/issues/39958), Could you have confirm this also fixed by [#45269](https://github.com/ansible/ansible/pull/45269) ? Thanks a lot!

**Comment by Fred-sun** [comment]
@zikalino  How about this issue? Could we can closed this? Thanks!

**Comment by Fred-sun** [comment]
@zikalino How about this issue going ahead ? Could we can closed this? Thanks!

**Comment by Akasurde** [comment]
!component =lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py

**Comment by Fred-sun** [comment]
duplicate of #39958

**Comment by Fred-sun** [comment]
@zikalino

**Comment by Fred-sun** [comment]
@zikalino @yungezz This issue seem resolve by #45269, Could we can close this? As #45269 mention new issue, It can be gave a new PR to track the key not exists issue. Could we can deal this issue like this? Thanks!

**Comment by Fred-sun** [comment]
@yungezz @zikalino @yuwzho Could we can closed this issue? It has fixed by #45269. The Standard SSD managed storage account type crashes issue has reported by issue #Azure/azure_preview_modules#182. It duplicate of the crash issue. I think we can close this issue and pay more attention on #Azure/azure_preview_modules#182. Thanks!

**Comment by Fred-sun** [comment]
kindly ping

**Comment by Fred-sun** [comment]
to_be_closed, The feature StandardSSD_LRS has been added into devel ansible.  Crash has also been fixed in the devel version. (Install devel ansible: "pip install git+https://github.com/ansible/ansible.git@devel ). Thanks!

**Comment by Fred-sun** [comment]
@yuwzho This issue similar with #39958, You can also close it! Thanks!

**Comment by Fred-sun** [comment]
advice closed, this duplicate with #39958(has been closed). It can be closed temporarily and reopened when needed. Thanks!

**Comment by yungezz** [comment]
@zikalino could you pls take a look at the issue?

**Comment by Fred-sun** [comment]
@zikalino Could we can closed this? Thanks!

**Comment by yuwzho** [comment]
close_me

As the DevTest Labs documentation said: If a user  is added as "DevTest Labs User" to a VM, he can only  start/stop/delete the VM, add/remove data disks to the VM only created by him. So After created by a contributor role, the vm cannot be stopped by a DevTest Labs User.

Reference: https://docs.microsoft.com/en-us/azure/lab-services/devtest-lab-add-devtest-user#actions-that-can-be-performed-in-each-role

BTW, we have added the DevTest Labs module, you can take a look or have a try: https://docs.ansible.com/ansible/2.8/modules/azure_rm_devtestlab_module.html
